data/metrics/src/testFixtures/java/tech/pegasys/teku/metrics/StubMetricsSystem.java

```diff
@@ -0,0 +1,78 @@
+/*
+ * Copyright 2020 ConsenSys AG.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not use this file except in compliance with
+ * the License. You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software distributed under the License is distributed on
+ * an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the License for the
+ * specific language governing permissions and limitations under the License.
+ */
+
+package tech.pegasys.teku.metrics;
+
+import java.util.Map;
+import java.util.Optional;
+import java.util.concurrent.ConcurrentHashMap;
+import java.util.function.DoubleSupplier;
+import org.hyperledger.besu.plugin.services.MetricsSystem;
+import org.hyperledger.besu.plugin.services.metrics.Counter;
+import org.hyperledger.besu.plugin.services.metrics.LabelledMetric;
+import org.hyperledger.besu.plugin.services.metrics.MetricCategory;
+import org.hyperledger.besu.plugin.services.metrics.OperationTimer;
+
+public class StubMetricsSystem implements MetricsSystem {
+
+  private Map<MetricCategory, Map<String, StubCounter>> counters = new ConcurrentHashMap<>();
+  private Map<MetricCategory, Map<String, StubGauge>> guages = new ConcurrentHashMap<>();
+
+  @Override
+  public LabelledMetric<Counter> createLabelledCounter(
+      final MetricCategory category,
+      final String name,
+      final String help,
+      final String... labelNames) {
+    return counters
```

Comment:
In line with how you did in `createGauge`, wouldn't it make more sense to check if a labeled counter with the same name exists? and throw an error if yes?

Reply:
No, the stub matches what the real system does - if you create the same counter twice it makes sense to just give you the same one as you're just incrementing each time.  That makes life a lot easier when counters are created in classes that get created multiple times etc.

For gauges though you can't just sum the values - if you have two with the same name you have to pick one to use and if you're recreating the same gauge all the time it can lead to a memory leak because the value supplier typically has a reference back to the parent class to access data and now the metrics system has a reference to the value supplier to the parent and everything it references can't be GC'd.